src/screens/OptionsAuthenticator/OptionsAuthenticatorScreen.tsx

```diff
@@ -57,16 +69,34 @@ class OptionsAuthenticatorScreen extends Component<Props> {
       });
   };
 
-  navigateToPair = () => {
-    const { authenticator, navigation } = this.props;
-
-    authenticator && navigation.navigate(Route.PairAuthenticator, { id: authenticator.id });
+  share = () => {
+    const { authenticator } = this.props;
+    Share.open({ message: authenticator?.publicKey });
   };
 
-  navigateToExport = () => {
-    const { authenticator, navigation } = this.props;
+  setName = (name: string) => this.setState({ name });
 
-    authenticator && navigation.navigate(Route.ExportAuthenticator, { id: authenticator.id });
+  get validationError(): string | undefined {
+    const { name } = this.state;
+    if (name?.length === 0) {
+      return i18n.authenticators.errors.noEmpty;
+    }
+    return;
+  }
+
+  saveNameAuthenticator = () => {
+    const { authenticator, updateAuthenticator } = this.props;
+    const { name } = this.state;
+    if (!!this.validationError) {
```

Comment:
instead of nesting condtion you can write
if (!!this.validationError || !authenticator) {
 return;
}

Reply:
done